libs/execution/include/hpx/parallel/executors/thread_execution.hpp

```diff
@@ -54,7 +54,7 @@ namespace hpx { namespace threads {
         lcos::local::futures_factory<result_type()> p(
             std::forward<Executor>(exec),
             util::deferred_call(std::forward<F>(f), std::forward<Ts>(ts)...));
-        p.apply();
+        p.apply(hpx::traits::get_function_annotation<typename hpx::util::decay<F>::type>::call(f));
```

Comment:
This will potentially access a moved-from object (see the `std::forward<F>(f)` above).

Reply:
> >     hpx::async(task_descriptor("function name", ...), &task_rddr, args...)
> What would be the difference between task_descriptor and the existing annotated_function (besides the name)?

2 principal differences. 
The first (which doesn't work, but I'll mention it anyway) is that currently we have
`hpx::async(annotated_function(name, func, args));`
and I would prefer
`hpx::async(annotated_function(name, func), args);`
but that's no good for lambdas unfortunately.

The second is that we currently have 
execution policies
executor parameters
function annotations
executors that add params without the user really knowing it is going on
Internally the stacksize, priority, hints, annotations are not handled very consistently and it seems like an execution_parameter that wrapped all of those up (plus new ones we have not yet added) 

A real difficulty is n executors like guided_pool and limiting (others?) that wrap other executors, can't easily forward this information onwards (partly caused by the fact that thread_executors have stacksize, but non thread executors don't) - there isn't a single place to get the info from and cleanly pass it around. 

So all I realy meant was that a 'task_descriptor' would be some kind of execution paramaters object tht could old lots of this kind of thing in one place and then the info could be managed more easily (similar to the thread_init struct that mikael added to the pools/schedulers etc).

Reply:
@biddisco Ok, understood. I think all of this falls into the domain of the future executor work we will have to do in any case. I'd suggest leaving things as they are for now.